src/async.rs

```diff
@@ -64,33 +66,210 @@ impl AsyncClient {
         Ok(Self::from_client(builder.base_url, client_builder.build()?))
     }
 
-    /// build an async client from the base url and [`Client`]
+    /// Build an async client from the base url and [`Client`]
     pub fn from_client(url: String, client: Client) -> Self {
         AsyncClient { url, client }
     }
 
-    /// Get a [`Transaction`] option given its [`Txid`]
-    pub async fn get_tx(&self, txid: &Txid) -> Result<Option<Transaction>, Error> {
-        let resp = self
-            .client
-            .get(&format!("{}/tx/{}/raw", self.url, txid))
-            .send()
-            .await?;
-
-        if let StatusCode::NOT_FOUND = resp.status() {
-            return Ok(None);
+    /// Make an HTTP GET request to given URL, deserializing to any `T` that
+    /// implement [`bitcoin::consensus::Decodable`].
+    ///
+    /// It should be used when requesting Esplora endpoints that can be directly
+    /// deserialized to native `rust-bitcoin` types, which implements
+    /// [`bitcoin::consensus::Decodable`] from `&[u8]`.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error either from the HTTP client, or the
+    /// [`bitcoin::consensus::Decodable`] deserialization.
+    async fn get_response<T: Decodable>(&self, path: &str) -> Result<T, Error> {
+        let url = format!("{}{}", self.url, path);
+        let response = self.client.get(url).send().await?;
+
+        match response.status().is_success() {
+            true => Ok(deserialize::<T>(&response.bytes().await?)?),
+            false => Err(Error::HttpResponse {
+                status: response.status().as_u16(),
+                message: response.text().await?,
+            }),
+        }
```

Comment:
style nit, not a blocker though

https://rust-lang.github.io/rust-clippy/master/index.html#/match_bool
```suggestion
        if response.status().is_success() {
            Ok(deserialize::<T>(&response.bytes().await?)?)
        } else {
            Err(Error::HttpResponse {
                status: response.status().as_u16(),
                message: response.text().await?,
            })
        }
```

Reply:
Thanks, I'll address it.

Reply:
FWIW I find this to be quite readable, but whatever you choose is fine

```rust
    if !response.status().is_success() {
        return Err(Error::HttpResponse {
            status: response.status().as_u16(),
            message: response.text().await?,
        });
    }

    Ok(deserialize::<T>(&response.bytes().await?)?)
```

Reply:
Cool! I did the update.

Reply:
I applied the same style to the other methods.